Formula/b/boring.rb

```diff
@@ -25,9 +25,18 @@ def post_install
     quiet_system "killall", "boring"
   end
 
+  def caveats
```

Comment:
I think we may use `generate_completions_from_executable` to do that, can you check?

Reply:
Hi, there is currently one issue with that: the tool's zsh completion script is written without the "#compdef" directive, and instead uses the `compdef` command directly, so that it can be sourced manually like the scripts for the other shells. So it doesn't currently work with the Homebrew completion system.

My idea was to keep the shell completion setup independent of Homebrew, so that the user has to add a single line to their shell init script to source completions, independent of the installation. I saw that other CLI tools like fzf are relying on the same approach.

What do you recommend doing? I could adapt the zsh completion script in the next minor release so that it works both with the directive and with manual sourcing, and add `generate_completions_from_executable`? This would make things slightly easier for users who have already setup the brew completion system, at the expense of having two (potentially incompatible?) ways of setting up shell completion... 

Thanks for your help!

Reply:
if not hassle, you can also check in the completion scripts (I see some projects were doing it). And we can just install the completion scripts in the install step.

Reply:
They are checked in -- and embedded into the binary at compile time. But I think it doesn't matter whether I install them like you proposed or with `generate_completions_from_executable`, the zsh completion system would still be unable to load the script, because it has to be sourced. Or am I misunderstanding something?

And more generally, do you advise against letting the user source the completions manually in their shell config file when using Homebrew? In that case I can also close this PR, do a bug fix update to make the zsh script work with the `#compdef` directive, and open a new PR. The other option would be to leave it as it currently is, as some other tools are doing the same. But not sure :)

Reply:
yeah, we'd prefer to install completions for users if possible, so it might be better to fix the zsh script and then use the checkedin completions.

no need to close this one and open a new one, you can just force update this PR to use the new release :)

Reply:
good point, will do this.

Reply:
sounds good 👍 

also need to squash all the commits :)

Reply:
done :)